test/no-function-overriding.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @fileoverview Test for no-function-overriding rule
+ * @author Nicolas Feignon <nfeignon@gmail.com>
+ */
+
+'use strict';
+
+var Solium = require ('solium');
+
+var userConfig = {
+    rules: {
+        "security/no-function-overriding": "error"
+    }
+};
+
+describe ('[RULE] no-function-overriding: Acceptances', function () {
+
+    it ('should accept contracts that don\'t override their inherited functions', function (done) {
+        var code = [
+            `
+                contract A {
+                    function foo();
+                    function ();
+                }
+                contract B is A {
+                    function bar();
+                    function ();
+                }
+            `,
+            `
+                contract A {
+                    function foo();
+                }
+                contract B is A {
+                    function bar();
+                    function();
+                }
+                contract C {
+                    function foo();
+                }
+            `,
+            `
+                contract A {
+                    function foo();
+                }
+                contract B is A {
+                    function bar();
+                    function foo(uint);
+                }
+            `,
+            `
+                contract A {
+                    function foo();
+                }
+                contract B is A {
+                    function foo(uint, string);
+                }
+                contract C is B {
+                    function foo(uint);
+                }
+            `
+        ];
+        var errors;
+
+        for (let expr of code) {
+            errors = Solium.lint (expr, userConfig);
+            errors.length.should.equal (0);
+        }
+
+        Solium.reset ();
+        done ();
+    });
+});
+
+describe ('[RULE] no-function-overriding: Rejections', function () {
+
+    it ('should reject contracts that override their inherited functions', function (done) {
+        var code = [
+            `
+                contract A {
+                    function foo();
+                }
+                contract B is A {
+                    function foo();
+                }
+            `,
+            `
+                contract A {
```

Comment:
Is it overriding if the overriding function has same param types but is without names? And the other way around?
```
contract A {
  function foo(uint x, string y) {}
}
contract B is A {
  function foo(uint, string) {}
}
```
If yes, please add tests for them.

Reply:
Only the number and type of parameters matters for function overriding.
I'll add a test :)